route/build/src/mpi_process.f90

```diff
@@ -676,133 +676,141 @@ subroutine mpi_restart(pid,           & ! input: proc id
+    if (onRoute(diffusiveWave)) then
+      call mpi_comm_molecule_state(pid, nNodes, comm,                        &
+                                   iens,                                     &
+                                   rch_per_proc(root:nNodes-1),              &
+                                   RCHSTA,                                   &
+                                   RCHSTA_trib,                              &
+                                   ixRch_order(rch_per_proc(root-1)+1:nRch), &
+                                   arth(1,1,rch_per_proc(pid)),              &
+                                   diffusiveWave,                            & ! routing method
+                                   scatter,                                  & ! communication type
+                                   ierr, message)
       if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+    end if
 
-      do iSeg = 1, rch_per_proc(pid)
-        RCHFLX_trib(iens,iSeg)%ROUTE(idxIRF)%REACH_VOL(iTbound-1) = vol_local(iSeg)
-      enddo
-    end do
-  endif
+    if (onRoute(impulseResponseFunc))then
+      call mpi_comm_irf_state(pid, nNodes, comm,                        &
+                              iens,                                     &
+                              rch_per_proc(root:nNodes-1),              &
+                              RCHFLX,                                   &
+                              RCHFLX_trib,                              &
+                              ixRch_order(rch_per_proc(root-1)+1:nRch), &
+                              arth(1,1,rch_per_proc(pid)),              &
+                              scatter,                                  & ! communication type
+                              ierr, message)
+      if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+      ! volume communication
+      do iTbound =1,nTbound
+        vol_global_tmp(:) = vol_global(:,iTbound)
+        call mpi_comm_single_flux(pid, nNodes, comm,                        &
+                                  vol_global_tmp,                           &
+                                  vol_local,                                &
+                                  rch_per_proc(root:nNodes-1),              &
+                                  ixRch_order(rch_per_proc(root-1)+1:nRch), &
+                                  arth(1,1,rch_per_proc(pid)),              &
+                                  scatter,                                  &
+                                  ierr, message)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+        do iSeg = 1, rch_per_proc(pid)
+          RCHFLX_trib(iens,iSeg)%ROUTE(idxIRF)%REACH_VOL(iTbound-1) = vol_local(iSeg)
+        enddo
+      end do
+    endif
+
+  end if ! (multProc)
 
   ! no need for the entire domain flux/state data strucure
   if (masterproc) then
```

Comment:
So this deallocation should always be done including the serial case? Just making sure...

Reply:
Yes, RCHFLX and  RCHSTA are deallocated safely.